drivers/pwm/Kconfig.max32

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 Analog Devices, Inc.
```

Comment:
Instead of 2023, should be 2023-2024, please apply it for all cases in this PR.

Reply:
Done